pkg/util/log/sampled.go

```diff
@@ -0,0 +1,37 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package log
+
+import (
+	"context"
+	"time"
+
+	"go.uber.org/atomic"
+)
+
+type SampledError struct {
+	Err     error
+	Sampler *Sampler
+}
+
+func (s SampledError) Error() string { return s.Err.Error() }
```

Comment:
Can we say here in the error message that it's sampled? It will be more obvious to debug issues, i.e.

```suggestion
func (s SampledError) Error() string {
    fmt.Sprintf("%s (sampled 1/%d)", s.Err, s.Sampler.freq)
}
```